Lib/email/quoprimime.py

```diff
@@ -296,5 +289,29 @@ def header_decode(s):
     quoted-printable (like =?iso-8859-1?q?Hello_World?=) -- please use
     the high level email.header class for that functionality.
     """
-    s = s.replace('_', ' ')
-    return re.sub(r'=[a-fA-F0-9]{2}', _unquote_match, s, flags=re.ASCII)
+    # Check for regex =[a-fA-F0-9]{2}
+    s = s.replace("_", " ")
+    
+    if '=' not in s:
+        return s
+    
+    result = ''
```

Comment:
Repeatedly appending to a string in a loop is O(n**2).  The standard idiom is to make a list of pieces (`result=[]`) and join after the loop.  I suspect that re.sub does the C equivalent.

In any case, I agree that replacing an re call with this much code seems dubious (a bad tradeoff), so closing this might be best.